src/components/CheckEditor/CheckEditor.test.tsx

```diff
@@ -0,0 +1,344 @@
+import React from 'react';
+import { render, screen, waitFor, act, within } from '@testing-library/react';
+import { Check, IpVersion, CheckType, DnsResponseCodes, ResponseMatchType } from 'types';
+import { CheckEditor } from './CheckEditor';
+import { getInstanceMock } from '../../datasource/__mocks__/DataSource';
+import userEvent from '@testing-library/user-event';
+import { InstanceContext } from 'components/InstanceContext';
+jest.setTimeout(20000);
+
+const defaultCheck = {
+  job: '',
+  target: '',
+  frequency: 60000,
+  timeout: 3000,
+  enabled: true,
+  labels: [],
+  probes: [],
+  settings: {
+    ping: {
+      ipVersion: IpVersion.V4,
+      dontFragment: false,
+    },
+  },
+} as Check;
+
+const getMinimumCheck = (overrides: any = {}) => ({
+  ...defaultCheck,
+  job: 'tacos',
+  target: 'burritos.com',
+  probes: [1],
+  ...overrides,
+});
+
+const onReturn = jest.fn();
+
+const selectCheckType = async (checkType: CheckType) => {
+  const checkTypeInput = await screen.findByText('PING');
+  userEvent.click(checkTypeInput);
+  const selectMenu = await screen.findByLabelText('Select options menu');
+  const option = await within(selectMenu).findByText(checkType.toUpperCase());
+  userEvent.click(option);
+  await screen.findByText(checkType.toUpperCase());
+};
+
+const openDnsValidations = async () => {
+  await selectCheckType(CheckType.DNS);
+  userEvent.click(await screen.findByText('Validation'));
+};
+
+const selectDnsResponseMatchType = async (responseMatch: ResponseMatchType) => {
+  const responseMatchInput = await screen.findByText('Validate Authority matches');
+  userEvent.click(responseMatchInput);
+  const options = await screen.findAllByText(`Validate ${responseMatch} matches`);
+  userEvent.click(options[options.length - 1]);
+};
+
+const submitForm = async () => {
+  const saveButton = await screen.findByRole('button', { name: 'Save' });
+  expect(saveButton).not.toBeDisabled();
+  userEvent.click(saveButton);
+  await waitFor(() => expect(onReturn).toHaveBeenCalledWith(true));
+};
+
+const renderCheckEditor = async ({ check = defaultCheck } = {}) => {
+  const instance = getInstanceMock();
+  render(
+    <InstanceContext.Provider value={{ instance: { api: instance }, loading: false }}>
+      <CheckEditor check={check} instance={instance} onReturn={onReturn} />
+    </InstanceContext.Provider>
+  );
+  await waitFor(() => expect(screen.getByText('Check Details')).toBeInTheDocument());
+  return instance;
+};
+
+it('renders without crashing', async () => {
```

Comment:
haha! Kinda implicitly covered by any other test as well :)

Reply:
True! I had this as a starting spot while working, and just didn't delete it. You think I should get rid of it?

Reply:
Getting rid would speed up tests by a second or two I guess! But doesn't really mater